mlflow/models/signature.py

```diff
@@ -257,7 +257,9 @@ def infer_signature(
                 schemas[key] = (
                     convert_dataclass_to_schema(data) if is_dataclass(data) else _infer_schema(data)
                 )
-            except Exception:
+            except Exception as e:
+                if isinstance(e, MlflowException) and "Pydantic objects" in e.message:
+                    raise
```

Comment:
can we define an error for this, and catch it here?

Reply:
Make sense